src/OpenTelemetry.Exporter.Jaeger/JaegerExporterOptions.cs

```diff
@@ -43,37 +46,63 @@ public class JaegerExporterOptions
 
         internal static readonly Func<HttpClient> DefaultHttpClientFactory = () => new HttpClient();
 
+        /// <summary>
+        /// Initializes a new instance of the <see cref="JaegerExporterOptions"/> class.
+        /// </summary>
         public JaegerExporterOptions()
+            : this(new ConfigurationBuilder().AddEnvironmentVariables().Build())
+        {
+        }
+
+        internal JaegerExporterOptions(IConfiguration configuration)
         {
-            if (EnvironmentVariableHelper.LoadString(OTelProtocolEnvVarKey, out string protocolEnvVar))
+            var protocol = configuration.GetValue<string>(OTelProtocolEnvVarKey, null);
+            if (!string.IsNullOrEmpty(protocol))
             {
-                if (JaegerExporterProtocolParser.TryParse(protocolEnvVar, out var protocol))
+                if (JaegerExporterProtocolParser.TryParse(protocol, out var parsedProtocol))
                 {
-                    this.Protocol = protocol;
+                    this.Protocol = parsedProtocol;
                 }
                 else
                 {
-                    throw new FormatException($"{OTelProtocolEnvVarKey} environment variable has an invalid value: '{protocolEnvVar}'");
+                    throw new FormatException($"{OTelProtocolEnvVarKey} environment variable has an invalid value: '{protocol}'");
                 }
             }
 
-            if (EnvironmentVariableHelper.LoadString(OTelAgentHostEnvVarKey, out string agentHostEnvVar))
+            var agentHost = configuration.GetValue<string>(OTelAgentHostEnvVarKey, null);
+            if (!string.IsNullOrEmpty(agentHost))
             {
-                this.AgentHost = agentHostEnvVar;
+                this.AgentHost = agentHost;
             }
 
-            if (EnvironmentVariableHelper.LoadNumeric(OTelAgentPortEnvVarKey, out int agentPortEnvVar))
+            var agentPort = configuration.GetValue<string>(OTelAgentPortEnvVarKey, null);
+            if (!string.IsNullOrEmpty(agentPort))
             {
-                this.AgentPort = agentPortEnvVar;
+                if (EnvironmentVariableHelper.LoadNumeric(OTelAgentPortEnvVarKey, agentPort, out int parsedAgentPort))
+                {
+                    this.AgentPort = parsedAgentPort;
+                }
             }
 
-            if (EnvironmentVariableHelper.LoadString(OTelEndpointEnvVarKey, out string endpointEnvVar)
-                && Uri.TryCreate(endpointEnvVar, UriKind.Absolute, out Uri endpoint))
+            var endpoint = configuration.GetValue<string>(OTelEndpointEnvVarKey, null);
```

Comment:
It would be better to provide `configuration.GetUri`

Reply:
Added a set of extensions off of `IConfiguration`